js/privatebin.js

```diff
@@ -3502,6 +3542,64 @@ jQuery.PrivateBin = (function($, RawDeflate) {
             $('#qrcode-display').html(qrCanvas);
         }
 
+        /**
+         * Send Email with current paste (URL).
+         *
+         * @name   TopNav.sendEmail
+         * @private
+         * @function
+         * @param  {Date|null} expirationDate date of expiration
+         * @param  {bool} isBurnafterreading whether it is burn after reading
+         */
+        function sendEmail(expirationDate, isBurnafterreading)
+        {
+            const EOL = '\n';
+            const BULLET = '  - ';
+            let emailBody = '';
+            const expirationDateRoundedToSecond = new Date(expirationDate);
+
+            // round down at least 30 seconds to make up for the delay of request
+            expirationDateRoundedToSecond.setUTCSeconds(
+                expirationDateRoundedToSecond.getUTCSeconds() - 30
+            );
+            expirationDateRoundedToSecond.setUTCSeconds(0);
+
+            const expirationDateString = window.confirm(
```

Comment:
If you want you can in Bootstrap also use a the little nicer bootstrap popup: https://stackoverflow.com/questions/22636819/confirm-deletion-using-bootstrap-3-modal-box

Or do something [with popovers](https://getbootstrap.com/docs/4.3/components/popovers/) like [this add-on did](http://bootstrap-confirmation.js.org/).

These are just some ideas...

Reply:
Hi,

I've tried it a little, but it seemed modals usually does not play nicely with mobile devices(shifting layout, page position or force scrolling).
Since the popup doesn't display any fancy content other than text, unlike the QR Code modal or password modal, I think the current implementation would be fine.